src/System.Private.CoreLib/shared/System/SpanHelpers.Byte.cs

```diff
@@ -199,10 +200,22 @@ public static unsafe int IndexOf(ref byte searchSpace, byte value, int length)
             IntPtr index = (IntPtr)0; // Use IntPtr for arithmetic to avoid unnecessary 64->32->64 truncations
             IntPtr nLength = (IntPtr)length;
 
-            if (Vector.IsHardwareAccelerated && length >= Vector<byte>.Count * 2)
+            if (Avx2.IsSupported || Sse2.IsSupported)
             {
-                int unaligned = (int)Unsafe.AsPointer(ref searchSpace) & (Vector<byte>.Count - 1);
-                nLength = (IntPtr)((Vector<byte>.Count - unaligned) & (Vector<byte>.Count - 1));
+                if (length >= Vector128<byte>.Count * 2)
+                {
+                    int unaligned = (int)Unsafe.AsPointer(ref searchSpace) & (Vector128<byte>.Count - 1);
```

Comment:
It would be really nice if the JIT would just optimize `Unsafe.AsPointer(ref searchSpace) % Vector128<byte>.Count`, or if we had a helper function for this type of code.

Reply:
@tannergooding - is there an open issue for this?

Reply:
Not sure, I'll take a look in a little bit (and will log one if we don't). It's probably worth noting that this optimization is only applicable for `unsigned` types (IIRC).

Reply:
Looks like the JIT already does this optimization for unsigned types. This is possibly just something that is normally missed due to most inputs being signed by default.

Reply:
Does it do it for an` int` that becomes a const when inlined? (as would be this case)

Reply:
I'll test that case after lunch. It seems like it would be valid to do when the constant is known to be positive.